### ISSUE TITLE
Spanner/ response bug without any row data in Spanner and IAM access to Spanner from CloudRun

### ISSUE THREAD

**Issue by Hirorou** [googleapis/google-cloud-go#6075]
**Client**

Spanner

**Environment**

- Spanner
  - doesn't have any row data in Spanner DB 
- Docker on CloudRun
  - doesn't have access permission to Spanner

**Go Environment**

- go version go1.18.1 darwin/amd64

**Code**

```go
package main

func main() {
  stmt := spanner.Statement{SQL: `SELECT column_data FROM table`}
  iter := s.client.Single().Query(ctx, stmt)
  var responseData []*entity.ResponseData
  defer iter.Stop()
  for {
      var columnA string
		
      row, err := iter.Next()
      if err == iterator.Done {
	  break
      }
      if err != nil {
	  return nil, err
      }
      if err := row.Columns(&columnA); err != nil {
	  return nil, err
      }
      data := &entity.ResponseData{
          ColumnA:   ColumnA,
      }
      responseData = append(responseData, data)
  }
return responseData
}
```

**Expected behavior**
Receive error response describing "Cloud Run doesn't have access permission to Spanner."

**Actual behavior**

However, I received "[]" data with Status code 200 from Cloud Run Server.
Therefore, the response message was hard for me to understand what a cause is.

**Screenshots**
<img width="1216" alt="Screen Shot 2022-05-22 at 17 57 23" src="https://user-images.githubusercontent.com/96335710/169687771-b473c5c3-e931-43ad-8d8f-9db607decdce.png">

**Additional context**
- gRPC
- gRPC Gateway
(I think they didn't affect this issue)

**Comment by kotofurumiya** [comment]
Hi, I'm Hirorou's colleague. I'm commenting some additional info, because this issue lacks some important environments.

- Cloud Run is not important. It just may be blame for SDK. 
- 200 OK is our application response. Spanner API nor SDK.
- data `[]` is generated by our application, not Spanner API nor SDK.

Accurate problem is,

- SDK's RowIterator.Next() returns `nil, nil` if runner instance has no appropriate permission.
- We think one of `row` or `err` must be non-nil, which means `row == nil && err != nil` or `row != nil && err != nil`.
- We want to know this behavior is by design, or bug. 

Sorry for my colleague's impoliteness. If you need more information, please mention to me.

**Comment by kotofurumiya** [comment]
Sorry, we confirmed this issue is fixed at latest(v1.32.0). We close it.
